includes/Checker/Checks/Enqueued_Styles_Scope_Check.php

```diff
@@ -0,0 +1,247 @@
+						array(
+							'code' => 'EnqueuedStylesScope.StyleLoadedInAllContext',
+							'file' => $plugin_style['path'],
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Gets the list of URLs to run this check for.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of URL strings (either full URLs or paths).
+	 *
+	 * @throws Exception Thrown when a post type URL cannot be retrieved.
+	 */
+	protected function get_urls() {
+		$urls = array( home_url( '/' ) );
+
+		foreach ( $this->get_viewable_post_types() as $post_type ) {
+			$args = array(
+				'posts_per_page'         => 1,
+				'post_type'              => $post_type,
+				'post_status'            => array( 'publish', 'inherit' ),
+				'ignore_sticky_posts'    => true,
+				'no_found_rows'          => true,
+				'lazy_load_term_meta'    => false,
+				'update_post_meta_cache' => false,
+				'update_post_term_cache' => false,
+			);
+
+			$the_query = new \WP_Query( $args );
+
+			if ( $the_query->have_posts() ) {
+				while ( $the_query->have_posts() ) {
+					$the_query->the_post();
+
+					$urls[] = get_permalink();
+
+					// Add post category link.
+					$id = get_the_ID();
+					if ( 'post' === get_post_type( $id ) ) {
+						$category = get_the_category( $id );
+						$urls[]   = get_category_link( $category[0]->term_id );
+					}
+				}
+			} else {
+				throw new Exception(
+					sprintf(
+						/* translators: %s: The Post Type name. */
+						__( 'Unable to retrieve post URL for post type: %s', 'plugin-check' ),
+						$post_type
+					)
+				);
+			}
+
+			/* Restore original Post Data */
+			wp_reset_postdata();
+		}
+
+		return $urls;
+	}
+
+	/**
+	 * Amends the given result by running the check for the given URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	protected function check_url( Check_Result $result ) {
+		// Reset the wp_styles instance.
+		unset( $GLOBALS['wp_styles'] );
+
+		/*
+		 * Run the 'wp_enqueue_script' action, wrapped in an output buffer in case of any callbacks printing styles
+		 * directly. This is discouraged, but some plugins or themes are still doing it.
+		 */
+		$wp_styles = wp_styles();
+		ob_start();
+			wp_enqueue_scripts();
+			$wp_styles->do_footer_items();
+		ob_get_clean();
+
+		foreach ( $wp_styles->done as $handle ) {
+			$style = $wp_styles->registered[ $handle ];
+
+			if ( ! $style->src || strpos( $style->src, $result->plugin()->url() ) !== 0 ) {
+				continue;
+			}
+
+			$style_path = str_replace( $result->plugin()->url(), $result->plugin()->path(), $style->src );
+
+			if ( isset( $this->plugin_style_count[ $handle ] ) ) {
+				$this->plugin_style_count[ $handle ] += 1;
+			} else {
+				$this->plugin_style_count[ $handle ] = 1;
+			}
```

Comment:
```suggestion
			if ( ! isset( $this->plugin_style_count[ $handle ] ) ) {
				$this->plugin_style_count[ $handle ] = 0;
			} 
			$this->plugin_style_count[ $handle ] += 1;
```

A little cleaner logic.